src/impl_parse_header.rs

```diff
@@ -0,0 +1,214 @@
+use crate::{constants::*, ResponseFlags};
+
+pub fn impl_parse_header(
+    data: &[u8],
+    start: usize,
```

Comment:
nit:  Same here with usize. I assume the data.len is not going to be greater than 65k?

Reply:
`&[u8]` should be able to be any size, to specify a fixed-size the type would be something like `&[u8;3]` I believe

Reply:
You can arguably configure a receive buffer of more than 64KB depending on your usage and avg value len...